python/glow/gwas/log_reg.py

```diff
@@ -106,18 +106,18 @@ def map_func(pdf_iterator):
     return genotype_df.mapInPandas(map_func, result_struct)
 
 
-@typechecked
-def _logistic_null_model_predictions(
-    y: NDArray[Any, Float],
-    X: NDArray[Any, Float],
-    y_mask: NDArray[Any, bool],
-    # nptying can't handle optional NDArrays, so don't verify the type for offset
-    offset
-) -> NDArray[Any, Float]:
+@dataclass
+class LogRegState:
+    inv_CtGammaC: NDArray[(Any, Any), Float]
+    gamma: NDArray[(Any, Any), Float]
+    Y_res: NDArray[(Any, Any), Float]
+
+
+def _logistic_null_model_predictions(y, X, mask, offset):
```

Comment:
It's usually a good idea to keep functions called within UDFs as trim as possible since they can be called many times. I don't believe typechecking is very expensive, but it can add up.